server/app/uk/gov/ons/addressIndex/server/modules/AddressIndexActions.scala

```diff
@@ -0,0 +1,114 @@
+package uk.gov.ons.addressIndex.server.modules
+
+import play.api.libs.json.Writes
+import play.api.mvc.Result
+import uk.gov.ons.addressIndex.crfscala.CrfScala.CrfTokenResult
+import uk.gov.ons.addressIndex.model.{BritishStandard7666, PostcodeAddressFile}
+import uk.gov.ons.addressIndex.model.db.index.{NationalAddressGazetteerAddresses, PostcodeAddressFileAddresses}
+import uk.gov.ons.addressIndex.model.server.response._
+import uk.gov.ons.addressIndex.server.controllers.PlayHelperController
+import uk.gov.ons.addressIndex.model.AddressScheme._
+import scala.concurrent.{ExecutionContext, Future}
+
+trait AddressIndexActions { self: AddressIndexCannedResponse with PlayHelperController =>
+
+  def esRepo: ElasticsearchRepository
+
+  /**
+    * required for handing of Futures.
+    */
+  implicit def ec: ExecutionContext
+
+  /**
+    * A simple type class which is used for distinction between query input types
+    */
+  sealed trait QueryInput[T] {
+    def input: T
+  }
+  case class UprnQueryInput(override val input: String) extends QueryInput[String]
+  case class AddressQueryInput(override val input: Seq[CrfTokenResult]) extends QueryInput[Seq[CrfTokenResult]]
+
+  /**
+    * @param tokens
+    * @return
+    */
+  def pafSearch(tokens: QueryInput[Seq[CrfTokenResult]]): Future[AddressBySearchResponseContainer] = {
+    esRepo queryPafAddresses tokens.input map { case PostcodeAddressFileAddresses(addresses, maxScore) =>
+      searchContainerTemplate(
+        tokens = tokens.input,
+        addresses = addresses map(AddressResponseAddress fromPafAddress maxScore),
+        total = addresses.size
+      )
+    }
+  }
+
+  /**
+    * @param tokens
+    * @return
+    */
+  def nagSearch(tokens: QueryInput[Seq[CrfTokenResult]]): Future[AddressBySearchResponseContainer] = {
+    esRepo queryNagAddresses tokens.input map { case NationalAddressGazetteerAddresses(addresses, maxScore) =>
+      searchContainerTemplate(
+        tokens = tokens.input,
+        addresses = addresses map(AddressResponseAddress fromNagAddress maxScore),
+        total = addresses.size
+      )
+    }
+  }
+
+  /**
+    * @param uprn
+    * @return
+    */
+  def uprnPafSearch(uprn: QueryInput[String]): Future[AddressByUprnResponseContainer] = {
+    esRepo queryPafUprn uprn.input map {
```

Comment:
I read quite a lot of scala style guides, I never saw any that advocates for "infix everywhere", did I miss something or is it just a personal preference?